src/components/modals/ScanModal.js

```diff
@@ -33,94 +38,157 @@ type DispatchProps = {
-          <TouchableHighlight style={styles.bottomButton} onPress={this._onToggleTorch} underlayColor={theme.secondaryButton}>
-            <View style={styles.bottomButtonTextWrap}>
-              <Ionicon style={styles.flashIcon} name="ios-flash" size={theme.rem(1.5)} />
-              <T style={styles.bottomButtonText}>{s.strings.fragment_send_flash}</T>
+        <>
+          <View style={styles.cameraContainer} onLayout={onLayoutCameraContainer}>
+            <RNCamera style={styles.cameraArea} captureAudio={false} flashMode={flashMode} onBarCodeRead={onBarCodeRead} type={RNCamera.Constants.Type.back} />
+          </View>
+
+          <QrPeephole width={peepholeWidth} height={peepholeHeight} holeSize={peepholeSize} holeTop={peepholeTop} />
+
+          <View style={{ ...styles.overlayContainer }}>
+            <View style={{ ...styles.headerContainer }}>
+              <SceneHeader withTopMargin title={title} underline />
             </View>
-          </TouchableHighlight>
-        </RNCamera>
+            <View
+              style={{
+                ...styles.peepholeSpace,
+                minHeight: peepholeSize
+              }}
+              onLayout={onLayoutPeepholeSpace}
+            />
+            <View style={styles.buttonsContainer}>
+              <TouchableHighlight onPress={onPressFlashIcon} underlayColor={theme.secondaryButton}>
+                <View style={styles.iconButtonContainer}>
+                  <Ionicon style={styles.icon} name="flash-outline" size={theme.rem(1.5)} />
+                  <T style={styles.iconButtonText}>{s.strings.fragment_send_flash}</T>
+                </View>
+              </TouchableHighlight>
+              <TouchableHighlight onPress={onPressAlbumIcon} underlayColor={theme.secondaryButton}>
+                <View style={styles.iconButtonContainer}>
+                  <Ionicon style={styles.icon} name="albums-outline" size={theme.rem(1.5)} />
+                  <T style={styles.iconButtonText}>{s.strings.fragment_send_album}</T>
+                </View>
+              </TouchableHighlight>
+            </View>
+          </View>
+        </>
       )
     }
 
     return (
-      <View style={styles.cameraArea}>
+      <View style={{ flex: 1, justifyContent: 'center' }}>
         <ActivityIndicator color={theme.primaryText} size="large" />
       </View>
     )
   }
 
-  close = () => {
-    this.props.bridge.resolve()
-  }
-
-  render() {
-    const { bridge, theme, title } = this.props
-
-    return (
-      <AirshipModal bridge={bridge} onCancel={this.close} borderRadius={theme.rem(1)} backgroundColor={theme.tileBackground}>
-        <ModalTitle>{title}</ModalTitle>
-        {this.renderCameraArea()}
-        <ModalCloseArrow onPress={this.close} />
-      </AirshipModal>
-    )
-  }
+  return (
+    <AirshipModal bridge={bridge} margin={[airshipMarginTop, 0, 0, 0]} padding={[0, 0, 0, 0]} backgroundColor={theme.modal} onCancel={close} overflow="hidden">
```

Comment:
You can use `margin={[airshipMarginTop, 0, 0]}` and `padding={0}`, and the other numbers will fill out, just like on the web.